### PR TITLE
[bitnami/postgresql-ha] feat: annotate pgpool deployment by checksum of custom user secrets

### PR DESCRIPTION
**Pull request by jjangga0214**
### Description of the change

https://github.com/bitnami/charts/issues/16396#issuecomment-1741981842

### Benefits

People will not create issues like #16396.

### Possible drawbacks

- When a user provides his/her own secret, the user still have to restart the deployment manually. But this is the obvious thing the user can expect, so not a drawback actually. To  automate such task, the user should use another tool like [Reloader](https://github.com/stakater/Reloader) by the way.
- The maintainers should sync the condition in the `if` statement between the *`deployment.yaml`* and *`custom-users-secrets.yaml`* when modified.

### Applicable issues

None, but related to #16396.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thank you for initiating this pull request. We appreciate your effort. Just a friendly reminder that it's important to sign your commits. Adding your signature certifies that you either authored the patch or have the necessary rights to contribute the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

Feel free to reach out if you have any questions or need assistance with the signing process.

**Comment by jjangga0214** [comment]
Signed it, thanks for the kind and prompt reply :)

File: bitnami/postgresql-ha/templates/pgpool/deployment.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ metadata:
     app.kubernetes.io/component: pgpool
   {{- if .Values.commonAnnotations }}
   annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- if and (not .Values.pgpool.customUsersSecret) .Values.pgpool.customUsers.usernames }}
+    checksum/custom-users-secrets: {{ include (print $.Template.BasePath "/custom-users-secrets.yaml") . | sha256sum }}
```
Comment (joancafom) [inline bitnami/postgresql-ha/templates/pgpool/deployment.yaml:18]: Isn't it `/pgpool/custom-users-secrets.yaml` the correct path?

```suggestion
    checksum/custom-users-secrets: {{ include (print $.Template.BasePath "/pgpool/custom-users-secrets.yaml") . | sha256sum }}
```

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for you PR! I left you a small comment as a review, could you please take a look at it when you have the chance?

**Comment by jjangga0214** [comment]
@joancafom Yes, you're right! Thanks :)

**Review by joancafom — APPROVED** [review]
LGTM! Thanks for your changes 😁

**Pull request merged**